packages/@aws-cdk/aws-iotevents/lib/state.ts

```diff
@@ -0,0 +1,44 @@
+import { Event } from './event';
+import { CfnDetectorModel } from './iotevents.generated';
+
+/**
+ * Properties for defining a state of a detector
+ */
+export interface StateProps {
+  /**
+   * The name of the state
+   */
+  readonly stateName: string
+
+  /**
+   * Specifies the actions that are performed when the state is entered and the `condition` is `TRUE`
+   *
+   * @default None
+   */
+  readonly onEnterEvents?: Event[]
```

Comment:
This property in not required, so it should not be implemented if I follow my roadmap.
> 1. implement DetectorModel and State with only required properties

However, this needed to be implemented. Because the detector model needed a reference to at least one AWS IoT Events input in order to deploy it.

Reply:
If that's the case, then we should validate that this condition is `true` before performing a deployment.